Move katana.loops operations to katana. Eliminate import cycle.

arthurp opened the pull request:
With: https://github.com/KatanaGraph/katana-enterprise/pull/1512

tylershunt commented:
> lgtm. Is there a difference in the generated documentation between putting this in the header and putting it in the source file?

I don't think so, this is more me trying to figure out what the function does an getting annoyed at having to open the cpp file 😁

tylershunt commented:
Mac build is failing looks like at the caching service. @ddn0  and @arthurp since they seem to have a handle on the implications there. For now I'm going to blow by it since the build itself seems to pass.
```
Post job cleanup.
/usr/bin/tar --use-compress-program zstd -T0 -cf cache.tzst -P -C /Users/runner/work/katana/katana --files-from manifest.txt
[warning]uploadChunk (start: 100663296, end: 134217727) failed: Cache service responded with 503
/Users/runner/work/_actions/actions/cache/v2/dist/save/index.js:3093
                        throw new Error(`Cache upload failed because file read failed with ${error.message}`);
                        ^

Error: Cache upload failed because file read failed with EBADF: bad file descriptor, read
    at ReadStream.<anonymous> (/Users/runner/work/_actions/actions/cache/v2/dist/save/index.js:3093:31)
    at ReadStream.emit (events.js:210:5)
    at internal/fs/streams.js:167:12
    at FSReqCallback.wrapper [as oncomplete] (fs.js:470:5)
```

arthurp commented:
> Mac CI is failing looks like at the caching service. @ddn0 and @arthurp since they seem to have a handle on the implications there. For now I'm going to blow by it since the build itself seems to pass.

This one is beyond me. I don't know. @ddn0?

ddn0 commented:
I was just assuming this is a transient failure. If it continues to happen, I can take a look.

arthurp committed:
python: Move katana.loops operations to katana. Eliminate import cycle.

arthurp committed:
python: Eliminate import cycle in _property_graph_numba.

arthurp committed:
python: Remove preemptive optimization of using __slots__

This started causing problems in sphinx for reasons I don't
understand. It worked in every other context. Shrug.

gliga commented on .github/workflows/python.yaml:
```diff
@@ -306,10 +306,10 @@ jobs:
           $HOME/build/python_env.sh python -c "import katana.analytics; katana.analytics.sssp;"
           . $HOME/build/python_env.sh
-          python -c "import katana.loops; katana.loops.for_each;"
+          python -c "import katana; katana.for_each;"
```
I like this a lot more

gliga reviewed (approved):
I like this change; it gets katana relevant operators closer to the user.  Hope this cycle removal will help us with Python+C++ integration.

arthurp commented:
Yea. Hopefully. I hope to continue this reorganization this month moving more things into katana and katana.local as appropriate.
